Content.Server/Fluids/EntitySystems/AbsorbentSystem.cs

```diff
@@ -185,14 +198,29 @@ private bool TryPuddleInteract(EntityUid user, EntityUid used, EntityUid target,
             return true;
         }
 
+        // hack to compensate loss due to fractions that would be periodical
+        // and make sure (available / Ratio) * Ratio is not less than available
+        // otherwise the absorber would very annoyingly never fill
+        // it's not necessary if the ratio is not periodical in a decimal base
+        if (PuddleSystem.EvaporationReagentRatio % 2 != 0 &&
+                PuddleSystem.EvaporationReagentRatio % 5 != 0){
+            available += FixedPoint2.Epsilon;
+        }
+        available *= PuddleSystem.EvaporationReagentRatio;
```

Comment:
to a more practical explanation: Prior to this change you would have the following water volumes in the mop, from the top of my head:
at 60 PickupAmount, that amounts to 20u water when wetting the mop;
while trying to mop a regular 20u puddle, you would use `20u / Ratio` water, and would wipe `20u / Ratio * Ratio` of puddle volume. Since these fractions can't be exactly represented in FixedPoint (or in ISO floating point), it happens that `20u / Ratio * Ratio < 20u`.
So after trying to mop 3 times you would have a mop with no water and something like 59.98 in dirt volume.
the size of the puddle doesn't really matter here, it would happen anyway due to this loss of precision in division.
If I add up 1 epsilon on the amount of puddle volume transfered at each wiping with the mop, I get `(20u / Ratio + 0.01) * Ratio > 20u`, hence now the mop can get full again after some mopping.